src/org/mozilla/javascript/NativeArray.java

```diff
@@ -1570,7 +1570,7 @@ else if (start < 0)
     private static Object iterativeMethod(Context cx, IdFunctionObject idFunctionObject, Scriptable scope,
                                           Scriptable thisObj, Object[] args)
     {
-        int id = idFunctionObject.methodId();
```

Comment:
I get what you're fixing here.

This case is already convoluted enough and the use of "Math.abs" is not going to be obvious to someone else using the code.

Can you add a comment to the effect that this works because at the very bottom of the file, we use negative numbers to represent the non-prototype versions of these functions and we want them all to work? Otherwise this looks good.

Reply:
Ok, you are right. Was not really obvious to me also. Will update this later today.

Reply:
Have added some inline comments, hope this makes this a bit more clear.